src/components/Form.jsx

```diff
@@ -74,4 +77,189 @@ export const AcceptanceForm = ({request, handleClose}) => {
+    const minutes = Array.from({ length: 60 }, (_, i) => i);  // Generates array [0, 1, 2, ..., 59]
+
+    const dropdownMenuStyle = {
+        maxHeight: '160px', // Approx. 5 rows (adjust as needed for row height)
+        overflowY: 'auto',
+    };
+
+    return (
+        <Container className="mt-4">
+            <Row className="justify-content-center">
+                <Col xs="auto">
+                    <DropdownButton 
+                        variant="secondary"
+                        id="dropdown-day"
+                        size="sm"
+                        title={selectedDay !== '' ? `${selectedDay} Days` : 'Day'}
+                        onSelect={(e) => {setSelectedDay(e); setTimer(e*24*60+selectedHour*60+selectedMinute*1)}}
+                    >
+                        <div style={dropdownMenuStyle}>
+                            {days.map((day) => (
+                                <Dropdown.Item key={day} eventKey={day}>
+                                    Day {day}
+                                </Dropdown.Item>
+                            ))}
+                        </div>
+                    </DropdownButton>
+                </Col>
+
+                <Col xs="auto">
+                    <DropdownButton
+                        variant="secondary"
+                        id="dropdown-hour"
+                        size="sm"
+                        title={selectedHour !== '' ? `${selectedHour} Hours` : 'Hour'}
+                        onSelect={(e) => {setSelectedHour(e); setTimer(selectedDay*24*60+e*60+selectedMinute*1)}}
+                    >
+                        <div style={dropdownMenuStyle}>
+                            {hours.map((hour) => (
+                                <Dropdown.Item key={hour} eventKey={hour}>
+                                    {hour < 10 ? `0${hour}` : hour} Hour(s)
+                                </Dropdown.Item>
+                            ))}
+                        </div>
+                    </DropdownButton>
+                </Col>
+
+                <Col xs="auto">
+                    <DropdownButton
+                        variant="secondary"
+                        id="dropdown-minute"
+                        size="sm"
+                        title={selectedMinute !== '' ? `${selectedMinute} Minutes` : 'Min'}
+                        onSelect={(e) => {setSelectedMinute(e); setTimer(selectedDay*24*60+selectedHour*60+e*1)}}
+                    >
+                        <div style={dropdownMenuStyle}>
+                            {minutes.map((minute) => (
+                                <Dropdown.Item key={minute} eventKey={minute}>
+                                    {minute < 10 ? `0${minute}` : minute} Minute(s)
+                                </Dropdown.Item>
+                            ))}
+                        </div>
+                    </DropdownButton>
```

Comment:
Nit: Looks like this is a lot of repeated code, could you make it reusable maybe